### DISCUSSION TITLE
advertising when connected?

### DISCUSSION THREAD

**Discussion by JavierRabba** [Zeni241/ESP32-NimbleBLE-For-Dummies#16] [General]
Hello,
First of all big thanks @Zeni241 for this repo. Its great.

I integrated these ideas into my ESP32 app and am currently facing an issue that I havent been able to resolve. When a connection is succesfull, the ESP stops advertising and it cant be found by other devices. This is an important feature to have in my app.

Now, when I try to run `bleprph_advertise` I get an error from `ble_gap_adv_start` with return code `6`

I dont find the myNewt documentation straight forward to interpret. Is error code `0x06` a `BLE_HS_ENOMEM` error? I dont know.
And if it where the case, is it possible to assign more stack to the nimble task?

or is it that it is not possible to advertise when connected?

I would appreciate your ideas, Thanks !!

**Comment by Zeni241** [comment]
Hi @JavierRabba 

As they say on [this](https://learn.adafruit.com/introduction-to-bluetooth-low-energy/gatt) page:

**_The most important thing to keep in mind with GATT and connections is that connections are exclusive. What is meant by that is that a BLE peripheral can only be connected to one central device (a mobile phone, etc.) at a time! As soon as a peripheral connects to a central device, it will stop advertising itself and other devices will no longer be able to see it or connect to it until the existing connection is broken._**

But if you want your sensor to communicate with multiple devices, then don't connect to it. Instead advertise the information you want to communicate. This way multiple devices can read this advertisement. 

Another solution could be to use [ESP-BLE-MESH](https://docs.espressif.com/projects/esp-idf/en/stable/esp32/api-guides/esp-ble-mesh/ble-mesh-index.html#esp-ble-mesh). 
. 
Another interesting solution could be ESP-NOW. ESP-NOW can support long-distance communication and consume less power.

Hope it helped.

**Reply by JavierRabba** [reply]
Thanks.
I found a way to connect and keep advertising, the solution was to restart advertisement after a connection was established with the non-connectable option

**Reply by Zeni241** [reply]
👌👍
